README.rst

```diff
@@ -939,6 +1009,194 @@ configuration status of a device changes, this ensures the check reacts
+in `OPENWISP_MONITORING_IPERF_CHECK_CONFIG <#OPENWISP_MONITORING_IPERF_CHECK_CONFIG>`_ 
+and the private key will be contained in the file ``private_not_protected.pem`` 
+which will be used with **--rsa-private-key-path** command option at iperf server.
+
+2. Create user credentials
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: shell
+
+   USER=iperfuser PASSWD=iperfpass
+   echo -n "{$USER}$PASSWD" | sha256sum | awk '{ print $1 }'
+   ----
+   ee17a7f98cc87a6424fb52682396b2b6c058e9ab70e946188faa0714905771d7 #This is the hash of "iperfuser"
+
+Add the above hash with username in ``credentials.csv``
+
+.. code-block:: shell
+
+   # file format: username,sha256
+   iperfuser,ee17a7f98cc87a6424fb52682396b2b6c058e9ab70e946188faa0714905771d7
+
+3. Now start the iperf server with auth options
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: shell
+
+   iperf3 -s --rsa-private-key-path ./private_not_protected.pem --authorized-users-path ./credentials.csv
+
+At client (openwrt device)
+##########################
+
+1. Install iperf3-ssl
+^^^^^^^^^^^^^^^^^^^^^
+
+Install `iperf3-ssl openwrt package <https://openwrt.org/packages/pkgdata/iperf3-ssl>`_ instead of normal
+`iperf3 openwrt package <https://openwrt.org/packages/pkgdata/iperf3>`_ which comes without any authentication.
+
+You may also check your installed **iperf3 openwrt package** features:
+
+.. code-block:: shell
+
+   root@vm-openwrt:~ iperf3 -v
+   iperf 3.7 (cJSON 1.5.2)
+   Linux vm-openwrt 4.14.171 #0 SMP Thu Feb 27 21:05:12 2020 x86_64
+   Optional features available: CPU affinity setting, IPv6 flow label, TCP congestion algorithm setting, 
+   sendfile / zerocopy, socket pacing, authentication # contains 'authentication'
+
+2. Configure iperf check auth parameters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now, add the following iperf authentication parameters 
+to `OPENWISP_MONITORING_IPERF_CHECK_CONFIG <#OPENWISP_MONITORING_IPERF_CHECK_CONFIG>`_ 
+in `openwisp settings <https://github.com/openwisp/openwisp-monitoring/blob/master/tests/openwisp2/settings.py>`_
+
+.. code-block:: python
+
+    OPENWISP_MONITORING_IPERF_CHECK_CONFIG = {
+    # All three parameters are required
+    'username': {'default': 'iperfuser'},
+    'password': {'default': 'iperfpass'},
+    # RSA public key without any headers
+    # ie. -----BEGIN PUBLIC KEY-----, -----BEGIN END KEY-----
+    'rsa_public_key': {
+        'default': """
+        MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAwuEm+iYrfSWJOupy6X3N 
+        dxZvUCxvmoL3uoGAs0O0Y32unUQrwcTIxudy38JSuCccD+k2Rf8S4WuZSiTxaoea 
+        6Du99YQGVZeY67uJ21SWFqWU+w6ONUj3TrNNWoICN7BXGLE2BbSBz9YaXefE3aqw 
+        GhEjQz364Itwm425vHn2MntSp0weWb4hUCjQUyyooRXPrFUGBOuY+VvAvMyAG4Uk 
+        msapnWnBSxXt7Tbb++A5XbOMdM2mwNYDEtkD5ksC/x3EVBrI9FvENsH9+u/8J9Mf 
+        2oPl4MnlCMY86MQypkeUn7eVWfDnseNky7TyC0/IgCXve/iaydCCFdkjyo1MTAA4 
+        BQIDAQAB
+        """
+    },
+   }
```

Comment:
We already have a setting to configure iperf, why are you adding a new one?

Can we compress everything in one setting?

Eg:

```python
OPENWISP_MONITORING_IPERF = {
    'a9734710-db30-46b0-a2fc-01f01046fe4f': [
        {
            'host': ['iperf1.openwisp.io', 'iperf2.openwisp.io'],
            'username': 'iperfuser',
            'password': 'iperfpassword',
            'rsa_public_key': (
"""
MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAwuEm+iYrfSWJOupy6X3N 
dxZvUCxvmoL3uoGAs0O0Y32unUQrwcTIxudy38JSuCccD+k2Rf8S4WuZSiTxaoea 
6Du99YQGVZeY67uJ21SWFqWU+w6ONUj3TrNNWoICN7BXGLE2BbSBz9YaXefE3aqw 
GhEjQz364Itwm425vHn2MntSp0weWb4hUCjQUyyooRXPrFUGBOuY+VvAvMyAG4Uk 
msapnWnBSxXt7Tbb++A5XbOMdM2mwNYDEtkD5ksC/x3EVBrI9FvENsH9+u/8J9Mf 
2oPl4MnlCMY86MQypkeUn7eVWfDnseNky7TyC0/IgCXve/iaydCCFdkjyo1MTAA4 
BQIDAQAB
""")
            'client_options': {
                'port': '6209',
                'udp': {
                    'bitrate': '20M'
                },
                'tcp': {
                    'bitrate': '0'
                }
            }
        }
    ],
}
``` 

Does it make sense?

Reply:
Yes, it would be better to have a single setting for `iperf check` :+1: